deeppavlov/dataset_readers/md_yaml_dialogs_reader.py

```diff
@@ -0,0 +1,388 @@
+                curr_action_text = cls._system_action2text(domain_knowledge, system_action_name)
+                system_action = {"speaker": cls._SYSTEM_SPEAKER_ID,
+                                 "text": curr_action_text,
+                                 "dialog_acts": [{"act": system_action_name, "slots": []}]}
+                if system_action_name.startswith("action"):
+                    system_action["db_result"] = {}
+
+                if curr_story_utters and curr_story_utters[-1]["speaker"] == cls._SYSTEM_SPEAKER_ID:
+                    # deal with consecutive system actions by inserting the last user replics in between
+                    last_user_utter = [u for u in reversed(curr_story_utters)
+                                       if u["speaker"] == cls._USER_SPEAKER_ID][0]
+                    curr_story_utters.append(last_user_utter)
+
+                curr_story_utters.append(system_action)
+
+        if not curr_story_bad:
+            stories_parsed[curr_story_title] = curr_story_utters
+        stories_parsed.pop(None)
+
+        tmp_f = tempfile.NamedTemporaryFile(delete=False, mode='w', encoding="utf-8")
+        for story_id, story in stories_parsed.items():
+            for replics in story:
+                print(json.dumps(replics), file=tmp_f)
+            print(file=tmp_f)
+        tmp_f.close()
+        gobot_formatted_stories = DSTC2DatasetReader._read_from_file(tmp_f.name, dialogs=dialogs)
+        os.remove(tmp_f.name)
+
+        log.debug(f"AFTER MLU_MD_DialogsDatasetReader._read_story(): "
+                  f"story_fpath={story_fpath}, "
+                  f"dialogs={dialogs}, "
+                  f"domain_knowledge={domain_knowledge}, "
+                  f"intent2slots2text={intent2slots2text}, "
+                  f"slot_name2text2value={slot_name2text2value}")
+
+        return gobot_formatted_stories
+
+    @classmethod
+    def _choose_slots_for_whom_exists_text(cls, intent2slots2text, slots_actual_values, user_action):
+        possible_keys = [k for k in intent2slots2text.keys() if user_action in k]
+        possible_keys = possible_keys + [user_action]
+        possible_keys = sorted(possible_keys, key=lambda action_s: action_s.count('+'))
+        for possible_action_key in possible_keys:
+            if intent2slots2text[possible_action_key].get(slots_actual_values):
+                slots_used_values = slots_actual_values
+                slots_to_exclude = []
+                return slots_to_exclude, slots_used_values, possible_action_key
+            else:
+                slots_lazy_key = set(e[0] for e in slots_actual_values)
+                slots_lazy_key -= {"intent"}
+                fake_keys = []
+                for known_key in intent2slots2text[possible_action_key].keys():
+                    if slots_lazy_key.issubset(set(e[0] for e in known_key)):
+                        fake_keys.append(known_key)
+                        break
+                fake_key = None
```

Comment:
redundant assignement since two lines below you assign `fake_key` again

Reply:
removed